### ISSUE TITLE
Documentation of vc4.qinc on VDW is incorrect

### ISSUE THREAD

**Issue by Seneral** [maazl/vc4asm#27]
Followed the documentation and used vdr_h32/vdr_v32 for DMA store, assuming that VDW and VDR used the same format. I kind of got it to work, but never seemed to fully understand it as there were weird behaviours (vdr_h32 with (1,0,0) for example works fine for writing 16x16 blocks, but has weird behaviour on 16x8 and others).

In short, the documentation should instead point to dma_h32/dma_v32, which are using the correct format. Also, they might be better named as vdw_h32/vdw_v32.

It is still not quite behaving as expected, although that is more likely to be my fault.

**Comment by Seneral** [comment]
Here's a probably unrelated issue I face, feel free to ignore it, I don't think it's a compiler related issue.

I want to write 16 units of length 32 into memory. What I get those 16 units of length 32 but it wraps back to 0 once it reaches 16 so I get two identical blocks of 16x16. I've checked that the VPM actually contains the content I want from 16-32 with two different methods.
16x16 writing, both with offset 0 and 16, works fine, and even doing 32x16 gives me the additional block I want, but of course in the wrong place.

mov vw_setup, vdw_setup_0(16, 32, dma_v32(0, 0));
![v16h32_photo](https://user-images.githubusercontent.com/12800840/77256603-924e5c00-6c6f-11ea-85b0-fac32a635f6b.png)
Notice the duplicate data despite asking for a continuous block.

mov vw_setup, vdw_setup_0(32, 16, dma_v32(0, 0));
![v32h16_photo](https://user-images.githubusercontent.com/12800840/77256607-94181f80-6c6f-11ea-9c70-5d3e2eec7b31.png)
Notice the continous block of memory requested is returned, but of course in the wrong format.

I tried both options for the blockmode in the stride setup as well. 
I reserved enough VPM memory for user programs (currently 64, the maximum).
QPU reports no errors, most noticeable no VPM store overflows or anything (which it does complain about when I purposefully overflow).
Any idea what I am doing wrong?

EDIT: Not a big deal anymore, as I won't need to write such large blocks anymore when I spread my workload on all QPUs, still something I haven't quite figured out.

**Comment by maazl** [comment]
1. The vc4asm documentation was indeed wrong. But the naming of the dma_... macros is chosen for compatibility with other toolsets, e.g. to compile the source code hello_fft.

2. The VideoCore IV documentation is somewhat misleading with respect to VDR/VDW and vertical/horizontal modes.
It is the VPM direction for consecutive elements of a row in memory. So if you choose _vertical_ mode the elements in a memory row are vertically aligned in VPM. The next row in memory will be in the next _column_ of VPM. Since VPM has only 16 columns it is quite obvious that you cannot write more than 16 rows this way.

**Comment by Seneral** [comment]
I think I got that, but I only want to write 16 memory rows, which should each be taking from one column, so according to your explanation it should fit without problems - it's the length parameter that makes issues. 
In vertical mode each column is 64 * 4 bytes long (at least addressible by a general purpose program provided you reserved 64 vectors of VPM memory beforehand, actual VPM size seems to be 192 vectors), so I would expect to be able to write 16 rows in memory, each taken from one of the 16 columns of length 64 *4 bytes. But past length 16 it wraps around. As far as I understand, this means it wraps at row 16 / 64 without any reason for each column.
However, in the 32 units of length 16 write it does what I expect it to do, it takes the first 16 units from the 16 columns in the first row (0/64) and then wraps, taking the next 16 units from the 16 column in the second row (16/64).
Since I wrote the data in horizontal mode, each unit in a row in horizontal mode should have the same value, gradually increasing with each row.

Of note is that I did manage to get it working as expected when I split up the VDW into two:
```
mov vw_setup, vdw_setup_0(16, 16, dma_v32(0, 0));
mov vw_setup, vdw_setup_0(16, 16, dma_v32(16, 0));
```
instead of
```
mov vw_setup, vdw_setup_0(16, 32, dma_v32(0, 0));
```
and then it does exactly what I wanted it to.
